### PR TITLE
Makefile: add tanzu-auth-controller-manager as a COMPONENT

### PR DESCRIPTION
**Pull request by ankeesler**
### What this PR does / why we need it

This PR ties the tanzu-auth controller into the build process used for the rest of the management components.

### Which issue(s) this PR fixes

None

### Describe testing done for PR

Not a whole lot...

### Release note

```release-note
NONE
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### Additional information

None

### CODE REVIEW

File: addons/pinniped/tanzu-auth-controller-manager/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# This Makefile is here to follow step 4 of this README:
+#   https://github.com/vmware-tanzu/tanzu-framework/blob/5cdba3181afbc31b05d4df51052fb47f3fcb5b92/docs/packages/add-new-package-to-package-repo.md
+
```
Comment (sabbey37) [inline addons/pinniped/tanzu-auth-controller-manager/Makefile:3]: Should we include the `common.mk` file?

Comment (ankeesler) [reply addons/pinniped/tanzu-auth-controller-manager/Makefile:3]: can you tell me why you are asking? i see that we can use the `ROOT_DIR` variable, but other than that i don't see anything else we would use from `common.mk` in our Makefile right now

Comment (sabbey37) [reply addons/pinniped/tanzu-auth-controller-manager/Makefile:3]: If there are not enough useful variables to make it worth including, then I'm ok leaving it out!

**Review by anujc25 — APPROVED** [review]
LGTM

**Review by yharish991 — APPROVED** [review]
LGTM!! Thanks for the changes :)

**Pull request merged**